nntrainer/src/databuffer.cpp

```diff
@@ -106,26 +175,36 @@ void DataBuffer::clear(BufferType type) {
     this->train_data_label.clear();
     this->cur_train_bufsize = 0;
     this->rest_train = max_train;
-    trainReadyFlag = false;
+    trainReadyFlag = DATA_NOT_READY;
     this->train_running = true;
+    if (validation[DATA_TRAIN])
```

Comment:
I think this can be changed to `if (validation[...] && true == train_thread.joinable())`

Reply:
fixed. Thanks.